WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt

```diff
@@ -264,14 +266,30 @@ class ProductListViewModel @AssistedInject constructor(
      * Resets the view state following a refresh
      */
     private fun resetViewState() {
+        // - If there are no products in default view, app shows empty view with its own add button, so hide
+        //   Add Product FAB.
+        //   - However, if there are no products as a result of searching or filtering, the empty view has no button,
+        //     so show Add Product FAB
+        // - Else If there's at least one product, show Add Product FAB.
+        val shouldShowAddProductButton =
+            if (_productList.value?.isEmpty() == true) {
+                when {
+                    viewState.query != null -> { true }
+                    productFilterOptions.isNotEmpty() -> { true }
+                    else -> { false }
+                }
+            } else {
+                true
+            }
```

Comment:
Nice, but IMO we can simplify further like this:
```suggestion
        val shouldShowAddProductButton = _productList.value?.isNotEmpty() == true ||
            viewState.query?.isNotEmpty() == true ||
            productFilterOptions.isNotEmpty()
```
WDYT?

Reply:
Thanks for the feedback. My main concern when writing that is that the conditionals here are not super intuitive ("hide button on empty view, but not if empty view is from search result", etc). I worried that in the future if I come back to this code, they will be hard to understand. So my goal is to make the if/else structure mimic the comment I wrote above it. It's more verbose but hopefully easier to follow.

I like the brevity in your proposed code, but from my beginner perspective, it takes me longer to follow and figure out what's happening. I'd like to keep the original code if possible. What do you think?

Reply:
Regarding your remark, when it comes to chained conditions, they are not hard to understand if you follow the logic, for example in our case, as it's an `or` relation, you just need to pick each condition, and then go to the next if it's false:
1. If the product list is not empty, show the add, else check 2.
2. if the search query is not null, show the add button, else check 3.
3. if the product filters is not empty, show the add button, otherwise hide it.
That's how I like to think about them 🙂.
But I'm not against keeping your version 🙂 , just if you do please remove the unneeded curly braces, as they are just noise in this case, I mean change it to something like this:
```kotlin
val shouldShowAddProductButton =
            if (_productList.value?.isEmpty() == true) {
                when {
                    viewState.query != null -> true
                    productFilterOptions.isNotEmpty() -> true
                    else -> false
                }
            } else {
                true
            }
```
And sorry for the many comments on the PR, it's helping us all learn from each other 🙂

Reply:
> And sorry for the many comments on the PR, it's helping us all learn from each other 🙂

No, no, don't be sorry at all :) I do appreciate and even need these sort of comments so I can learn further.

I agree with the braces, have removed them.